### PR TITLE
Release/1.1.0

### PR DESCRIPTION
**Pull request by paulboocock**
Hey,
So this PR is mainly aimed at bumping dependencies in the main Snowplow.Tracker so we no longer use System.Net.Http v4.3.0.
I've also added support for an in memory queue, so we're not always forcing LiteDB, and added subjects per event rather than only at the tracker level.

I've published 1.1.0-beta.1 for both Snowplow.Tracker and Snowplow.Tracker.PlatformExtensions (which wraps 1.1.0-beta.1). The publishing flow is a little odd here as I need to publish Snowplow.Tracker first before I can update PlatformExtensions so you end up with two releases per PR with a commit in between. I've tags each "release" but I'll also tag "1.1.0" on master (for snowplow/snowplow) after merging.

I haven't updated the CHANGELOG yet as I'll make a full Prepare for release commit once approved and I've deleted the temporarty beta commits.

TODO
- [x] Action PR feedback
- [x] Delete Temp commits
- [x] Update CHANGELOG
- [x] Prepare for release with final Snowplow.Tracker version numbers
- [x] Release Snowplow.Tracker 1.1.0
- [x] Update PlatformExtensions to use Snowplow.Tracker 1.1.0
- [x] Prepare for release with final Snowplow.Tracker.PlatformExtensions version numbers
- [x] Release PlatformExtensions 1.1.0
- [x] Merge PR

### CODE REVIEW

**Review by adatzer — APPROVED** [review]
Looks great! 
Just some nit-picking and a couple of questions in the comments.
For the record, locally from the Demo.Console, all events good in Micro!

Also opened this minor test [issue](https://github.com/snowplow/snowplow-dotnet-tracker/issues/112), but it was not part of this PR, so just mentioning in case it fits :)

File: .github/workflows/deploy_extensions.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      run: msbuild Snowplow.Tracker.PlatformExtensions.sln /property:Configuration=Release
+
+    - name: Pack Snowplow.Tracker.PlatformExtensions
+      run: nuget pack .\NuGet\Snowplow.Tracker.PlatformExtensions.nuspec -Symbols -SymbolPackageFormat snupkg
+
+    - name: Check nupkg version
+      uses: andstor/file-existence-action@v1
+      with:
+        files: "Snowplow.Tracker.PlatformExtensions.${{ github.event.inputs.version }}.nupkg, Snowplow.Tracker.PlatformExtensions.${{ github.event.inputs.version }}.snupkg"
+        allow_failure: true
```
[the remark this answers is not in the document]
Comment (paulboocock) [reply .github/workflows/deploy_extensions.yml:39]: Yeah it does. Theres an open issue on this repo that it's a terrible name! It's basically backwards 🤦

File: Snowplow.Tracker.PlatformExtensions/Snowplow.Tracker.PlatformExtensions.Abstractions/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@
 [assembly: AssemblyConfiguration("")]
 [assembly: AssemblyCompany("Snowplow Analytics LTD")]
 [assembly: AssemblyProduct("Snowplow.Tracker.PlatformExtensions.Abstractions")]
-[assembly: AssemblyCopyright("Copyright © 2017")]
+[assembly: AssemblyCopyright("Copyright © 2021")]
```
[the remark this answers is not in the document]
Comment (paulboocock) [reply Snowplow.Tracker.PlatformExtensions/Snowplow.Tracker.PlatformExtensions.Abstractions/Properties/AssemblyInfo.cs:14]: No harm in doing so! It should definitely be on the Dotnet461 project, the others are less important as we don't release them.

File: Snowplow.Tracker.Tests/Queues/InMemoryBlockingQueue.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
+﻿/*
+ * PersistentBlockingQueueTest.cs
```
[the remark this answers is not in the document]
Comment (paulboocock) [reply Snowplow.Tracker.Tests/Queues/InMemoryBlockingQueue.cs:2]: Doesn't need it in c# land but good spot, it's good practice.

File: Snowplow.Tracker/Queues/InMemoryBlockingQueue.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        private readonly object _queueLock = new object();
+
+        private ConcurrentDictionary<Guid, Payload> _collection;
+
+        /// <summary>
+        /// Create a persistent blocking queue - store queue across executions
+        /// </summary>
+        /// <param name="s">Interface to storage</param>
+        /// <param name="payloadToString">Serialization method</param>
+        public InMemoryBlockingQueue()
```
[the remark this answers is not in the document]
Comment (paulboocock) [reply Snowplow.Tracker/Queues/InMemoryBlockingQueue.cs:38]: Good spot. This is why comments are terrible right? Far too easy to forget to update them 🤦

**Pull request merged**